modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -105,9 +116,58 @@ const preventDeletingSummary = (editor: Editor): void => {
         e.preventDefault();
         removeNode(editor, node);
         CaretFinder.firstPositionIn(nextNode).each(setCaretToPosition(editor));
+      } else if (isSafari && isSummary(node)) {
+        // TINY-9951: Safari has a bug where upon pressing Backspace/Delete when the caret is directly within the summary,
+        // all content is removed and the caret is prevented from being placed back into the summary.
+        if (isMacOS && (e.altKey || e.metaKey) || e.ctrlKey) {
+          // Allow ranged deletion by keyboard shortcuts natively, pad summary and relocate caret on keyup if needed
```

Comment:
I can't seem to get this to work in my Safari it always removes the summary and doesn't pad it on ctrl-backspace.
I think you can mimic ctrl delete with using the non-standard modify API.
```
  // Deletes the previous word (uses left since I think delete is always in logical order not visual order)
  editor.getSel().modify('extend', 'left', 'word');
  editor.getDoc().execCommand('Delete');
```

Reply:
But it does weird things if you do this at the first or last character of a details body. So probably needs additional checks. So might not be something we can squeeze in here.

Reply:
@spocke That's a great suggestion. I have reimplemented the workaround for shortcut ranged deletion using `Selection.modify()` and it works a lot better. I resolved most of the issues by performing the ranged deletion override within the temporary span container we created. I made improvements to the tests regarding shortcut ranged deletion as well.

Also, Ctrl + Backspace is supposed to be a Windows/Linux shortcut, and Alt + Backspace is the macOS equivalent. I addressed this by adding `!isMacOS && e.ctrlKey` to the condition.